src/rcc.rs

```diff
@@ -45,6 +46,87 @@ pub struct Rcc {
     pub cfgr: CFGR,
 }
 
+impl Rcc {
```

Comment:
I don't know if this is the right place to add this configuration. I'm not sure about this, though. This is not really used in the rest of the `rcc` module as of now. The structure of the rcc module could take a redesign anyway, but I have a bad gut-feeling about this approach. Maybe you could explain this part a bit more, to convince me.

Reply:
I was going back and forth with this as well. Is there a benefit to defining each register as a field within the Rcc struct? Initially, I was going to follow the current style, as well as the [stm32f1xx-hal](https://github.com/stm32-rs/stm32f1xx-hal/blob/master/src/rcc.rs) style and do something like the following:

```
impl RccExt for RCC {
    fn constrain(self) -> Rcc {
        Rcc {
            ahb: AHB { _0: () },
            apb1: APB1 { _0: () },
            apb2: APB2 { _0: () },
            cfgr: CFGR {
                hse: None,
                hclk: None,
                pclk1: None,
                pclk2: None,
                sysclk: None,
                adcclk: None,
            },
            bkp: BKP { _0: () }, // add backup domain register struct as field
            cr: CR { _0: () } // add Control register struct as field
            // etc etc
        }
    }
}
```
but because I did not fully understand why it was written that way, I went with what I considered to be more straightforward.

Reply:
This "reexports" parts of the rcc periphery, so that it can be passed to other modules and is guarded by the rust owner-ship rules instead of having to access it via unsafe. 

The A.. types resemble the peripheral clock dividers, which have to be assessed to e.g. enable the spi peripheral clock, so that it is working. A somewhat related issue could help you understand it: https://github.com/stm32-rs/stm32f3xx-hal/issues/37

Reply:
That makes sense. I changed it so the registers needed for RTC initialization are reexported.